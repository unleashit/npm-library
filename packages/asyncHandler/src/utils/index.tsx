const isEmpty = (data: any): boolean => {
  if (Array.isArray(data)) {
    return !data.length;
  }
  if (typeof data === 'object') {
    return !Object.keys(data).length;
  }
  return !data;
};

export { isEmpty };
