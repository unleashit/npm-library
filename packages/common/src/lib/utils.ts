interface CSSModule {
  [key: string]: string;
}

const isCSSModule = (theme: CSSModule | undefined, prop: string, fallback: string): string => {
  return (theme && theme[prop]) ? `${theme[prop]}` : fallback;
};

export {
  isCSSModule
}
