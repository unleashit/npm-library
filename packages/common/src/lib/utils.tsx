import * as React from 'react';

const isCSSModule = (themeProp: string | undefined, fallback: string): string => {
  return themeProp || fallback;
};

const returnComponentFormat = (
  // eslint-disable-next-line @typescript-eslint/ban-types
  C: React.ReactNode | Function,
  attrs: Record<string, unknown>,
): any => {
  if (React.isValidElement(C)) {
    return C;
  }
  if (typeof C === 'function') {
    return <C {...attrs} />;
  }
  throw new Error(
    'Not a valid component. Please supply a component or valid React element.',
  );
};

const sentenceCase = (str: string): string => {
  return (
    str.charAt(0).toUpperCase() +
    str
      .slice(1)
      .split(/(?=[A-Z])/)
      .join(' ')
  );
};

export { isCSSModule, returnComponentFormat, sentenceCase };
