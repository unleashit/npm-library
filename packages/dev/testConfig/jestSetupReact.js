/* eslint-disable import/no-extraneous-dependencies */
const React = require('react');
const Enzyme = require('enzyme');
const Adapter = require('@cfaester/enzyme-adapter-react-18');

// jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

// Make Enzyme functions and React available in all test files without importing
global.React = React;
global.shallow = Enzyme.shallow;
global.render = Enzyme.render;
global.mount = Enzyme.mount;

// global mocks
// check for window since Jest environment is node in certain tests
if (typeof window !== 'undefined') {
  window.alert = jest.fn((msg) => console.log(msg));
}

// if (typeof global !== 'undefined') {
//   global.alert = jest.fn((msg) => console.log(msg));
// }
