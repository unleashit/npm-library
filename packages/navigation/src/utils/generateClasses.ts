import { NavigationProps } from '../navigation';
import { NavLinkProps } from '../NavLink';

type Direction = NavigationProps['direction'];
type Template = NavigationProps['template'];
type AuthLink = NavLinkProps['authLink'];
type ClsName = (title: string) => string;

// container--vertical
const isVertical = (dir: Direction, clsName: ClsName): string =>
  dir === 'vert' || dir === 'vertical' ? ` ${clsName('vertical')}` : '';

const isTemplate = (template: Template, clsName: ClsName): string => {
  let classes = '';
  switch (template) {
    case 'clean':
      classes = ` ${clsName('clean')}`;
      break;
    case 'dark-buttons':
      classes = ` ${clsName('darkButtons')}`;
      break;
    case 'light-buttons':
      classes = ` ${clsName('lightButtons')}`;
      break;
    case 'none':
      classes = ` ${clsName('none')}`;
      break;
    default:
      classes = '';
  }
  return classes;
};

export const addTemplateClasses = (
  template: Template,
  direction: Direction,
  clsName: ClsName,
): string =>
  `${isTemplate(template, clsName)}${isVertical(direction, clsName)}`;

export const getAuthLinkClass = (
  authLink: AuthLink,
  clsName: ClsName,
): string => {
  if (authLink && ['login', 'logout', 'signup'].includes(authLink)) {
    return ` ${clsName(`${authLink}Link`)}`;
  }

  return '';
};

export const mapArrayToClasses = <T extends string[] | undefined>(
  classes: T,
): string => (classes && classes.length ? ` ${classes.join(' ')}` : '');
