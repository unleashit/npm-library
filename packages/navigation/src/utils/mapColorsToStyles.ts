import { CSSProperties } from 'react';
import { utils } from '@unleashit/common';

// export interface Colors {
//   light?: CSSProperties['color'];
//   lightDarker5?: CSSProperties['color'];
//   lightDarker10?: CSSProperties['color'];
//   lightDarker15?: CSSProperties['color'];
//   dark?: CSSProperties['color'];
//   darkLighter5?: CSSProperties['color'];
//   darkLighter10?: CSSProperties['color'];
//   darkLighter15?: CSSProperties['color'];
//   textLight?: CSSProperties['color'];
//   textDark?: CSSProperties['color'];
// }

// export const mapColorsToStyles = (colors?: Colors) => {
//   const styles: Record<string, string> = {};
//   if (!colors) return styles;
//
//   colors.light && (styles['--unl-navigation-light'] = colors.light);
//   colors.dark && (styles['--unl-navigation-dark'] = colors.dark);
//   colors.textLight &&
//     (styles['--unl-navigation-text-light'] = colors.textLight);
//   colors.textDark && (styles['--unl-navigation-text-dark'] = colors.textDark);
//   colors.lightDarker5 &&
//     (styles['--unl-navigation-light-darker-5'] = colors.lightDarker5);
//   colors.lightDarker10 &&
//     (styles['--unl-navigation-light-darker-10'] = colors.lightDarker10);
//   colors.lightDarker15 &&
//     (styles['--unl-navigation-light-darker-15'] = colors.lightDarker15);
//   colors.darkLighter5 &&
//     (styles['--unl-navigation-dark-lighter-5'] = colors.darkLighter5);
//   colors.darkLighter10 &&
//     (styles['--unl-navigation-dark-lighter-10'] = colors.darkLighter10);
//   colors.darkLighter15 &&
//     (styles['--unl-navigation-dark-lighter-15'] = colors.darkLighter15);
//   return styles;
// };

const colorKeys = [
  'light',
  'dark',
  'textLight',
  'textDark',
  'lightDarker5',
  'lightDarker10',
  'lightDarker15',
  'darkLighter5',
  'darkLighter10',
  'darkLighter15',
] as const;

export type Colors = {
  [key in (typeof colorKeys)[number]]?: CSSProperties['color'];
};

export const mapColorsToStyles = (colors?: Colors) => {
  if (!colors) return {};

  const styles: Partial<Record<string, string>> = {};

  colorKeys.forEach((key) => {
    if (colors[key]) {
      const kebabCaseKey = `--unl-navigation-${utils.stringToKebab(key)}`;

      styles[kebabCaseKey] = colors[key];
    }
  });

  return styles;
};
