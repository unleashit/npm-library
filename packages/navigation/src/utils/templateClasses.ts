import { isCSSModule } from '@unleashit/common';

import { Props, Style } from '../index';
import { NavLinkExtraProps } from '../NavLink';

const isVertical = (dir: Props['direction'], style: Style): string => {
  return dir === 'vert' || dir === 'vertical'
    ? ` ${isCSSModule(style.vertical, 'unl-navigation__container--vertical')}`
    : '';
};

const isTemplate = (template: Props['template'], style: Style): string => {
  let classes = '';
  switch (template) {
    case 'plain':
      classes = ` ${isCSSModule(style.plain, 'unl-navigation__container--plain')}`;
      break;
    case 'clean':
      classes = ` ${isCSSModule(style.clean, 'unl-navigation__container--clean')}`;
      break;
    case 'dark-buttons':
      classes = ` ${isCSSModule(
        style.darkButtons,
        'unl-navigation__container--dark-btns',
      )}`;
      break;
    case 'light-buttons':
      classes = ` ${isCSSModule(
        style.lightButtons,
        'unl-navigation__container--light-btns',
      )}`;
      break;
    default:
      classes = '';
  }
  return classes;
};

export const addTemplateClasses = (
  template: Props['template'],
  direction: Props['direction'],
  theme: Style,
): string => {
  return `${isTemplate(template, theme)}${isVertical(direction, theme)}`;
};

export const getAuthClasses = (
  authLink: NavLinkExtraProps['authLink'],
  theme: Style,
): string => {
  if (authLink === 'login') {
    return ` ${isCSSModule(theme.loginLink, 'unl-navigation__login-link')}`;
  }
  if (authLink === 'logout') {
    return ` ${isCSSModule(theme.logoutLink, 'unl-navigation__logout-link')}`;
  }
  if (authLink === 'signup') {
    return ` ${isCSSModule(theme.signupLink, 'unl-navigation__signup-link')}`;
  }
  return '';
};
