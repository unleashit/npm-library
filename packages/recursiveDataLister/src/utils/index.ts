export const getChildTag = (tag: string): keyof JSX.IntrinsicElements => {
  if (tag === 'ul' || tag === 'ol') return 'li';
  return 'div';
};

export const isObjectNotArray = (
  elem: Record<string, unknown> | any[],
): boolean => typeof elem === 'object' && !Array.isArray(elem);

// const isObjectNotNull = (elem: any): boolean =>
//   !!elem && typeof elem === 'object' && !Array.isArray(elem);

export const isDate = (elem: unknown): boolean =>
  Object.prototype.toString.call(elem) === '[object Date]' ||
  // eslint-disable-next-line no-restricted-globals
  (typeof elem === 'string' && !isNaN(new Date(elem) as any));

// const isStringDate = (elem: unknown): boolean =>
//   // eslint-disable-next-line no-restricted-globals
//   typeof elem === 'string' && !isNaN(new Date(elem) as any);
export const isObjectNotDate = (
  elem: Record<string, unknown> | any[],
): boolean => typeof elem === 'object' && !isDate(elem);

export type DateFormat = (val: Date) => string | number;

export const handleDate = (elem: Date, cb: DateFormat): string | number =>
  cb(elem);

export const isPrimitive = (elem: unknown): boolean =>
  typeof elem === 'string' ||
  typeof elem === 'number' ||
  typeof elem === 'boolean';
