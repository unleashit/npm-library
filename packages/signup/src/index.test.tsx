import { mount, ReactWrapper } from 'enzyme';
import * as React from 'react';
import { act } from 'react-dom/test-utils';

// eslint-disable-next-line import/no-relative-packages
import { changeVal, nextTick } from '../../../testConfig/utils';
import SignupContainer, { mapDefaultValues, Props } from '.';

describe('<SignupContainer />', () => {
  let wrapper: ReactWrapper;
  let props: Props;

  beforeEach(() => {
    props = {
      signupHandler: () => Promise.resolve({ success: true }),
      onSuccess: () => jest.fn(),
    };

    wrapper = mount(<SignupContainer {...props} />);
  });

  it('renders without crashing', () => {
    expect(wrapper.find('.unl-signup__container')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  describe('validation', () => {
    it('fields can be updated and validated', async () => {
      await act(async () => {
        wrapper
          .find('input[name="email"]')
          .simulate('change', changeVal('email', 'test@test.com'));
        wrapper
          .find('input[name="password"]')
          .simulate('change', changeVal('password', '12345678'));
      });
      wrapper.update();
      expect(wrapper.find('input[name="email"]').props().value).toEqual('test@test.com');
      const errors = wrapper.find('.errorMessage');
      expect(errors).toHaveLength(0);
    });

    it('displays correct email validation errors', async () => {
      const emailInput = wrapper.find('input[name="email"]');
      const passwordInput = wrapper.find('input[name="password"]');
      const passwordConfirmInput = wrapper.find('input[name="passwordConfirm"]');
      const form = wrapper.find('form');

      // email can't be empty
      await act(async () => {
        passwordInput.simulate('change', changeVal('password', 'goodpassword'));
        form.simulate('submit', { preventDefault: () => undefined });
      });
      wrapper.update();
      let emailError = wrapper.find('.unl-signup__error-message').at(0);
      expect(emailError.text()).toEqual('email is a required field');

      // email must be a valid email
      await act(async () => {
        emailInput.simulate('change', changeVal('email', 'bad email'));
        passwordInput.simulate('change', changeVal('password', 'goodpassword'));
        passwordConfirmInput.simulate('change', changeVal('password', 'goodpassword'));
        await nextTick();
        form.simulate('submit', { preventDefault: () => undefined });
      });
      wrapper.update();
      emailError = wrapper.find('.unl-signup__error-message').at(0);
      expect(emailError.text()).toEqual('email must be a valid email');
    });

    it('displays correct password validation errors', async () => {
      const emailInput = wrapper.find('input[name="email"]');
      const passwordInput = wrapper.find('input[name="password"]');
      const form = wrapper.find('form');

      // password field can't be empty
      await act(async () => {
        emailInput.simulate('change', changeVal('email', 'good@email.com'));
        await nextTick();
        form.simulate('submit', { preventDefault: () => undefined });
      });
      wrapper.update();
      let passwordError = wrapper.find('.unl-signup__error-message').at(0);
      expect(passwordError.text()).toEqual('password is a required field');

      // password must be at least 8 chars
      await act(async () => {
        emailInput.simulate('change', changeVal('email', 'good@email.com'));
        passwordInput.simulate('change', changeVal('password', '123'));
        await nextTick();
        form.simulate('submit', { preventDefault: () => undefined });
      });
      wrapper.update();
      passwordError = wrapper.find('.unl-signup__error-message').at(0);
      expect(passwordError.text()).toEqual('password must be at least 8 characters');
    });
  });

  describe('helper functions', () => {
    it('mapDefaultValues() returns the right object', () => {
      const fields: any = [
        {
          element: 'input',
          type: 'text',
          name: 'email',
          label: 'Email',
          defaultValue: 'something',
        },
        {
          element: 'input',
          type: 'password',
          name: 'password',
          label: 'Password',
        },
      ];
      /*
       *  return shape of mapDefaultValues()
       *   {
       *     [name]: defaultValue || ''
       *   }
       * */
      const { email, password } = mapDefaultValues(fields);

      expect(email).toEqual('something');
      expect(password).toEqual('');
    });
  });
});
