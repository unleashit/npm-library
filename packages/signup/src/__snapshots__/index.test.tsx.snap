// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupContainer /> renders without crashing 1`] = `
<WithFormik(Signup)
  onSuccess={[Function]}
  signupHandler={[Function]}
>
  <Formik
    handleSubmit={[Function]}
    initialValues={
      Object {
        "email": "",
        "password": "",
        "passwordConfirm": "",
        "serverAuth": "",
      }
    }
    onSubmit={[Function]}
    onSuccess={[Function]}
    signupHandler={[Function]}
    validationSchema={[Function]}
  >
    <Signup
      dirty={false}
      errors={Object {}}
      getFieldHelpers={[Function]}
      getFieldMeta={[Function]}
      getFieldProps={[Function]}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialErrors={Object {}}
      initialTouched={Object {}}
      initialValues={
        Object {
          "email": "",
          "password": "",
          "passwordConfirm": "",
          "serverAuth": "",
        }
      }
      isSubmitting={false}
      isValid={true}
      isValidating={false}
      onSuccess={[Function]}
      registerField={[Function]}
      resetForm={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      signupHandler={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      validateOnMount={false}
      values={
        Object {
          "email": "",
          "password": "",
          "passwordConfirm": "",
          "serverAuth": "",
        }
      }
    >
      <div
        className="unl-signup__container"
      >
        <SignupHeader
          linkComponent={[Function]}
          linkComponentHrefAttr="href"
          loginUrl="/login"
          theme={Object {}}
        >
          <div
            className="unl-signup__header"
          >
            <h2>
              Sign Up
            </h2>
            <p>
              Already have an account?
               
              <DefaultLinkComponent
                href="/login"
              >
                <a
                  href="/login"
                >
                  Login
                </a>
              </DefaultLinkComponent>
            </p>
          </div>
        </SignupHeader>
        <Form
          className="unl-signup__form"
        >
          <form
            action="#"
            className="unl-signup__form"
            onReset={[Function]}
            onSubmit={[Function]}
          >
            <Field
              component={[Function]}
              componentName="signup"
              cssModule={Object {}}
              name="email"
              type="text"
            >
              <CustomInput
                componentName="signup"
                cssModule={Object {}}
                field={
                  Object {
                    "name": "email",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "value": "",
                  }
                }
                form={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "getFieldHelpers": [Function],
                    "getFieldMeta": [Function],
                    "getFieldProps": [Function],
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialErrors": Object {},
                    "initialStatus": undefined,
                    "initialTouched": Object {},
                    "initialValues": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "status": undefined,
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validateOnMount": false,
                    "values": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                  }
                }
                type="text"
              >
                <div
                  className="unl-signup__form-group"
                >
                  <label
                    className="unl-signup__label"
                    htmlFor="signup-form-email"
                  >
                    Email
                  </label>
                  <input
                    className="unl-signup__input "
                    id="signup-form-email"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </CustomInput>
            </Field>
            <Field
              component={[Function]}
              componentName="signup"
              cssModule={Object {}}
              name="password"
              type="password"
            >
              <CustomInput
                componentName="signup"
                cssModule={Object {}}
                field={
                  Object {
                    "name": "password",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "value": "",
                  }
                }
                form={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "getFieldHelpers": [Function],
                    "getFieldMeta": [Function],
                    "getFieldProps": [Function],
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialErrors": Object {},
                    "initialStatus": undefined,
                    "initialTouched": Object {},
                    "initialValues": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "status": undefined,
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validateOnMount": false,
                    "values": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                  }
                }
                type="password"
              >
                <div
                  className="unl-signup__form-group"
                >
                  <label
                    className="unl-signup__label"
                    htmlFor="signup-form-password"
                  >
                    Password
                  </label>
                  <input
                    className="unl-signup__input "
                    id="signup-form-password"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </CustomInput>
            </Field>
            <Field
              component={[Function]}
              componentName="signup"
              cssModule={Object {}}
              name="passwordConfirm"
              type="password"
            >
              <CustomInput
                componentName="signup"
                cssModule={Object {}}
                field={
                  Object {
                    "name": "passwordConfirm",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "value": "",
                  }
                }
                form={
                  Object {
                    "dirty": false,
                    "errors": Object {},
                    "getFieldHelpers": [Function],
                    "getFieldMeta": [Function],
                    "getFieldProps": [Function],
                    "handleBlur": [Function],
                    "handleChange": [Function],
                    "handleReset": [Function],
                    "handleSubmit": [Function],
                    "initialErrors": Object {},
                    "initialStatus": undefined,
                    "initialTouched": Object {},
                    "initialValues": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                    "isSubmitting": false,
                    "isValid": true,
                    "isValidating": false,
                    "registerField": [Function],
                    "resetForm": [Function],
                    "setErrors": [Function],
                    "setFieldError": [Function],
                    "setFieldTouched": [Function],
                    "setFieldValue": [Function],
                    "setFormikState": [Function],
                    "setStatus": [Function],
                    "setSubmitting": [Function],
                    "setTouched": [Function],
                    "setValues": [Function],
                    "status": undefined,
                    "submitCount": 0,
                    "submitForm": [Function],
                    "touched": Object {},
                    "unregisterField": [Function],
                    "validateField": [Function],
                    "validateForm": [Function],
                    "validateOnBlur": true,
                    "validateOnChange": true,
                    "validateOnMount": false,
                    "values": Object {
                      "email": "",
                      "password": "",
                      "passwordConfirm": "",
                      "serverAuth": "",
                    },
                  }
                }
                type="password"
              >
                <div
                  className="unl-signup__form-group"
                >
                  <label
                    className="unl-signup__label"
                    htmlFor="signup-form-passwordConfirm"
                  >
                    Password Confirm
                  </label>
                  <input
                    className="unl-signup__input "
                    id="signup-form-passwordConfirm"
                    name="passwordConfirm"
                    onBlur={[Function]}
                    onChange={[Function]}
                    type="password"
                    value=""
                  />
                </div>
              </CustomInput>
            </Field>
            <button
              className="unl-signup__button"
              type="submit"
            >
              Sign Up
            </button>
          </form>
        </Form>
      </div>
    </Signup>
  </Formik>
</WithFormik(Signup)>
`;
